docs/content/en/docs/concepts/architecture/components/scheduler/_index.md

```diff
@@ -14,3 +14,152 @@ The Lifecycle Scheduler can trigger events such as
 deployment, testing, and remediation at specific times or intervals.
 The Keptn Scheduler guarantees that Pods are initiated only after
 the Pre-Deployment checks are completed.
+
+## Architecture of Keptn Scheduler
+
+To integrate the `Keptn Lifecycle Scheduler` into a Kubernetes cluster,
+a Kubernetes manifest is applied with specific **Keptn annotations**.
+These annotations identify the workload and specify the
+pre-deployment and post-deployment tasks and evaluations
+associated with it.
+
+## How it works
+
+The `Keptn Scheduler`, integrated into the `Keptn Lifecycle Operator`,
+which is equipped with multiple controllers for managing
+**Keptn Custom Resource Definitions (CRDs)** and a powerful
+**Mutating Webhook**.
+Its core responsibility is to orchestrate the scheduling of Pods, taking
+into account pre-deployment checks to ensure optimal deployment success.
+
+Let's dive into its architectural details:
+
+- **Scheduler Flow**:
+After applying a Kubernetes Manifest annotated with Keptn-specific
+annotations to the cluster, the Mutating Webhook injects the Keptn Scheduler.
+The Scheduler then receives Kubernetes Events for
+Pre-Deployment, which are sent to the event stream.
+The Event Controller watches for these events and triggers a
+Kubernetes Job to fulfill the Pre-Deployment tasks.
+Once the Pre-Deployment phase is complete,
+the Keptn Scheduler schedules the Pod for deployment.
```

Comment:
This is not exactly how it works, the scheduler is triggered every time a new workload is added, it will wait for the workload instance pre-evaluation status to be successful before binding each pod to a node. Please refer to this https://github.com/keptn/lifecycle-toolkit/blob/main/README.md#scheduler